pytorch_lightning/trainer/training_tricks.py

```diff
@@ -80,3 +86,175 @@ def configure_accumulated_gradients(self, accumulate_grad_batches):
             self.accumulation_scheduler = GradientAccumulationScheduler(schedule)
         else:
             raise TypeError("Gradient accumulation supports only int and dict types")
+
+    def scale_batch_size(self,
+                         model: LightningModule,
+                         mode: str = 'power',
+                         n_step_per_try: int = 3,
+                         init_val: int = 2,
+                         n_max_try: int = 25):
+        r"""
+        Will iteratively try to find the largest batch size for a given model
+        that does not not give an out of memory (OOM) error
+
+        Args:
+            model: Model to fit.
+
+            mode: string setting the search mode. Either `power` or `binsearch`.
+                If mode is `power` we keep multiplying the batch size by 2, until
+                we get an OOM error. If mode is 'binsearch', we will initially
+                also keep multiplying by 2 and after encountering an OOM error
+                do a binary search between the last succeded batch size and the
+                batch size that failed.
+
+            n_step_per_try: number of steps to run with a given batch size.
+                Idealy 1 should be enough to test if a OOM error occurs,
+                however in practise a few is needed
```

Comment:
```suggestion
            n_step_per_try: number of steps to run with a given batch size.
                Idealy 1 should be enough to test if a OOM error occurs,
                however in practise a few are needed
```
number args accross the PL codebase typically start with `num_*`. Consider changing to `num_steps_per_try`? :)